### PR TITLE
Bag(s) of Holding (things)

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
This PR is a balance pass and nerf to bags of holding.
Regular BoH design now requires bluespace 8, but it can hold up to 45 items (+10).
Two new types of bag of holding added, each requiring bluespace 7 (same as current) that do specialist things.

- Dufflebag of Holding: This has a combined weight class of 18, but allows storage of huge and bulky items, excluding other bags. (Example use: 2 huge items with 6 spare slots for normal items.) This bag has a green colour scheme.
- Satchel of Holding: This has a combined weight class of 30 but with no slowdown. This bag does not have the ability to hold bulky/huge items. This bag has a blue colour scheme.

TODO:
Sprites that aren't recolours.

## Why It's Good For The Game
Storage creep is real, currently, BoHs are common as muck and do literally everything. The only malus they have is interrupting teleports, and that's such a rare situation that it does not affect the majority of users.
Most people agree that BoH is currently overpowered.
There is currently no chance for decisions by players about what sort of use they want from their bag, it's just one big one and done.
This PR means that bags of holding will not be as strong as they have been, and encourages player choice for their needs. Want to carry fire axes about in your bag? Take the bag of large things. (No, you can't put bags in this to get more storage space either.)
Want to carry lots of regular stuff? Take the many things bag, but it can't hold as many as before!

Simply making the BoH much harder to obtain seemed like an unappealing route for me to nerf it, so offering less powerful alternatives while keeping the good item in the rare event of bluespace 8 tech seemed ideal.

## Changelog
:cl:
add: Added two new specialised types of "Bag of Holding" requiring bluespace 7.
tweak: Regular "Bag of Holding" now requires bluespace 8.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 1]
```diff
@@ -85,6 +85,22 @@
 	var/dist = max((current_size - 2), 1)
 	explosion(loc, dist, (dist * 2), (dist * 4))
 
+/obj/item/storage/backpack/holding/huge_items
+	name = "Bag of Holding (large things)"
+	desc = "A backpack that opens into a localized pocket of Blue Space. This variant can hold huge items, at the cost of capacity"
+	icon_state = "holdingpacklarge"
+	item_state = "holdingpacklarge"
+	max_combined_w_class = 12
```
Comment (S34NW) [inline code/game/objects/items/weapons/storage/backpack.dm:93]:
```suggestion
	max_combined_w_class = 18
```
This was not meant to be 12, you should be able to store 2 huge items with room to spare.

**Comment by Gatchapod** [comment]
Imagine sacrificing your drip to wear the powergamer's bag.

And seriously, I like the idea of adding yet another use for bluespace anomaly cores. Picking between a Phazon, reactive teleport armour and BoHs is a good idea. Indifferent towards replacement bags, I'm not sacrificing my looks.

**Comment by LightFire53** [comment]
Satchel of Holding and Duffle bag of holding?

File: code/modules/research/designs/bluespace_designs.dm [hunk 2]
```diff
@@ -13,17 +13,17 @@
...
+	req_tech = list("bluespace" = 8, "materials" = 6, "engineering" = 7, "plasmatech" = 6)
 	build_type = PROTOLATHE
 	materials = list(MAT_GOLD = 3000, MAT_DIAMOND = 1500, MAT_URANIUM = 250, MAT_BLUESPACE = 2000)
 	build_path = /obj/item/storage/backpack/holding
 	category = list("Bluespace")
 
 /datum/design/bag_holding_large
-	name = "Bag of Holding (large things)"
-	desc = "A backpack that opens into a localized pocket of Blue Space. This variant can hold huge items, at the cost of capacity"
+	name = "Duffelbag of Holding"
```
Comment (hal9000PR) [inline code/modules/research/designs/bluespace_designs.dm:25]: Surely these should have different sprites? I feel like they need different sprites.

Comment (hal9000PR) [reply code/modules/research/designs/bluespace_designs.dm:25]: nevermind

**Review by Kyep — COMMENTED** [review]
I'm skeptical that the BoH needs nerfing.
Forcing people to be more picky about what they put in their bag... doesn't seem like it is particularly fun, nor does it seem as great for balance as people seem to think it is.
Perhaps the best argument for nerfing BoHs is that it stops people walking around with 10 guns in their bag, then firing each of them in turn at a blob. But... it doesn't actually stop them. It just makes them use a locker to do the same thing. And if the guns don't have infinite ammo cycling through 10 guns isn't actually that big of a problem as now you have 10 guns to reload/recharge.
Nerfing the BoH is one of those things that seems like it will force "interesting choices" but in reality it won't, it just makes carrying stuff a little more tedious.

**Comment by Wojz** [comment]
Bags of holding can hold bags which can hold normal items, which makes for even stupider amounts of items carried than boxes let you do with normal bags (wonder if/when that's being looked at). Being able to conceal something like fireaxe/chainsaw is, to put it bluntly, kind of insane, considering the Powergamer's Bag also allows miners to carry 5 different OP swords, all the traitor gear they want, and medical supplies on top of that in case something penetrated their bullshitium plating.

While yes, people will be able to drag lockers full of guns still, that is at least potentially counterable for a blob, via blobbernauts and crew's own incompetency. That locker can be stolen, pushed away, get stuck for long enough that blob snakes to it and blocks it, and such.

**Comment by LordShekelstein** [comment]
> I'm skeptical that the BoH needs nerfing.
Right, allow me to change your mind on this then. 
> Forcing people to be more picky about what they put in their bag... doesn't seem like it is particularly fun, nor does it seem as great for balance as people seem to think it is.
They already need to be incredibly picky about what the put into their bag already though, bag of holding or not. Inventory management is a huge part of this game, hence why the bag of holding is such a powerful item. 
> Perhaps the best argument for nerfing BoHs is that it stops people walking around with 10 guns in their bag, then firing each of them in turn at a blob. But... it doesn't actually stop them. It just makes them use a locker to do the same thing. And if the guns don't have infinite ammo cycling through 10 guns isn't actually that big of a problem as now you have 10 guns to reload/recharge.
What? Carrying around an entire department's worth of meds, 8 AEGS, and miner gamer loot in a BoH is completely different from dragging around a locker with 10 guns, having to manually open it, click the sprite behind you, and spam at the blob. One is far clunkier than the other. 
> Nerfing the BoH is one of those things that seems like it will force "interesting choices" but in reality it won't, it just makes carrying stuff a little more tedious.
No, this was absolutely not the scope of the PR (at least from my read, S34N can correct me if I'm wrong). The intent was that the BoH was an overpowered item that required nerfing and thusly putting it in a position where it was more difficult to get and required the sacrifice of several potentially more powerful items was good for balance.

**Comment by hal9000PR** [comment]
Currently. The bag of holding allows you to: 
Carry various items that are balanced around not being carried, for example, spellblade, spectral sword, fire axes, arrythmatic knife (sorta defeats the slowdown point if you can just store it in a bag), hardsuits and space suits.

And, hold a shit tonn of various items,  7 AEGs, a medical belt filled with meds, a tool belt filled with tools, and a full space suit. That isn't even considering using bags and boxes to increase your storage capacity.

For the massive con of teleporting randomly, which can actually be a boon.



One issue I have with this nerf is that people will just take a bag of holding big things put a duffel bag in it + two bulkey items. Effectively having the same capacity as an old bag of holding. Minor complaint however.

**Comment by S34NW** [comment]
> 
> One issue I have with this nerf is that people will just take a bag of holding big things put a duffel bag in it + two bulkey items. Effectively having the same capacity as an old bag of holding. Minor complaint however.

You are unable to put bags of any sort in the bag of holding big things

**Comment by ppi13** [comment]
I disagree with this PR, it all comes down to being able to carry guns and medical supplies, these are the arguments for, I mean, apply a speed potion to two duffel bags and anti-drop implant one of the bags, and bam, you can still carry 10 guns and a departments worth of items, heck, drag a third duffelbag, you need to be physically stunned somehow, or have someone else grab that.
Not to speak of using bluespace lockers or body bags to carry items with, sure you can't instantly get every single item, but you have it so your priority items are in the backpack you wear.
So the real inconvenience here is the limited amount of bulky items.

Not only that, these new bags are harder to get then the old bags, requiring higher materials and engineering to get for something that is worse then the original? Ontop of that you nerf the original BoH storage capacity AND make it harder to get?

And making the original require a bluespace anomaly, which you might have two off per day?

Ontop of that, the lesser bags can't even hold backpacks

This is not nerfing an item, this is Killing it, and "Storage Creep" is not a valid enough reason for that.

How is this really a good thing, how does this make it really better, this does not improve anything.

Also, you state making it more difficult to get is is an unappealing, thing, and then you go on and make it more difficult in all but material cost and plasmatech

**Comment by Spacemanspark** [comment]
Everyone: Discussion about bag of holding balance

Me: If the satchel of holding doesn't have access to the 'switch strap side' verb I'm down voting this PR. >:^(

**Comment by S34NW** [comment]
> 
> 
> I disagree with this PR, it all comes down to being able to carry guns and medical supplies, these are the arguments for, I mean, apply a speed potion to two duffel bags and anti-drop implant one of the bags, and bam, you can still carry 10 guns and a departments worth of items, heck, drag a third duffelbag, you need to be physically stunned somehow, or have someone else grab that.

So your argument is: BoH is fine because you can ANTI-DROP two speed potioned duffel bags and you get the same result.
This is ludicrous, if anything this shows how terribly broken the system currently is, you should not have this much carry capacity in an item that is handed out like free samples in a supermarket.

> Not to speak of using bluespace lockers or body bags to carry items with, sure you can't instantly get every single item, but you have it so your priority items are in the backpack you wear.
> So the real inconvenience here is the limited amount of bulky items.

Bluespace lockers themselves are an issue, we calculated using BS lockers and BoH's, you could carry 22 **_million_** small items.

> 
> Not only that, these new bags are harder to get then the old bags, requiring higher materials and engineering to get for something that is worse then the original? Ontop of that you nerf the original BoH storage capacity AND make it harder to get?

The mat and engineering tweaks were actually done in error, thank you for pointing it out. The original BoH isn't nerfed, its BUFFED. +10 storage slots.

>
> And making the original require a bluespace anomaly, which you might have two off per day?

Yes! That's the point! If they were easily available the whole point of the nerf would be for nothing.

>
> Ontop of that, the lesser bags can't even hold backpacks

Again, this is the point. If I permitted this, they would just be bags of holding with more steps to get to your items.

>
> This is not nerfing an item, this is Killing it, and "Storage Creep" is not a valid enough reason for that.

The original item is essentially dead, yes. That's why we have two replacement bags that require you to actually use more than one neuron when selecting.

> Also, you state making it more difficult to get is is an unappealing, thing, and then you go on and make it more difficult in all but material cost and plasmatech

Yes, but I provide ALTERNATIVES. Just making the original rarer would be lame, it would go the way of the phazon mech. Introducing less-powerful alternatives that do specialised things is much much better design than "haha bag hold everything".

File: code/modules/research/designs/bluespace_designs.dm [hunk 3]
```diff
@@ -13,14 +13,34 @@
...
 	build_type = PROTOLATHE
 	materials = list(MAT_GOLD = 3000, MAT_DIAMOND = 1500, MAT_URANIUM = 250, MAT_BLUESPACE = 2000)
 	build_path = /obj/item/storage/backpack/holding
 	category = list("Bluespace")
 
+/datum/design/bag_holding_large
+	name = "Duffelbag of Holding"
+	desc = "A backpack that opens into a localized pocket of Blue Space. This variant can hold huge items, at the cost of capacity."
+	id = "bag_holding_large"
+	req_tech = list("bluespace" = 7, "materials" = 6, "engineering" = 7, "plasmatech" = 6)
```
Comment (S34NW) [inline code/modules/research/designs/bluespace_designs.dm:28]:
```suggestion
	req_tech = list("bluespace" = 7, "materials" = 5, "engineering" = 5, "plasmatech" = 6)
```

File: code/modules/research/designs/bluespace_designs.dm [hunk 4]
```diff
@@ -13,14 +13,34 @@
...
+	build_type = PROTOLATHE
+	materials = list(MAT_GOLD = 3000, MAT_DIAMOND = 1500, MAT_URANIUM = 250, MAT_BLUESPACE = 4000)
+	build_path = /obj/item/storage/backpack/holding/huge_items
+	category = list("Bluespace")
+
+/datum/design/bag_holding_many
+	name = "Satchel of Holding"
+	desc = "A backpack that opens into a localized pocket of Blue Space. This variant can hold more items, at the cost of item size limitations."
+	id = "bag_holding_many"
+	req_tech = list("bluespace" = 7, "materials" = 6, "engineering" = 7, "plasmatech" = 6)
```
Comment (S34NW) [inline code/modules/research/designs/bluespace_designs.dm:38]:
```suggestion
	req_tech = list("bluespace" = 7, "materials" = 5, "engineering" = 5, "plasmatech" = 6)
```

**Review by S34NW — COMMENTED** [review]
these edits were copy-pasta'd in error

**Comment by variableundefined** [comment]
Passed 30 days without head or maintainer support. 

Closing.

**Pull request closed without merging**